darwin/datatypes.py

```diff
@@ -19,6 +21,38 @@
 ErrorHandler = Callable[[int, str], None]
 
 
+@dataclass(frozen=True)
+class StringDataSource:
+    """
+    Represents a source from a ``StringData`` class.
+    """
+
+    #: The id of the source.
+    id: str
+
+    #: The range of the source.
+    ranges: Optional[List[Tuple[int, int]]]
+
+    @staticmethod
+    def parse(source: Dict[str, Union[str, Optional[List[Tuple[int, int]]]]]) -> "StringDataSource":
```

Comment:
I think - but I'm not 100% sure - you can "solve" this by importing `from __future__ import annotations`, and then
```suggestion
    def parse(source: Dict[str, Union[str, Optional[List[Tuple[int, int]]]]]) -> StringDataSource:
```

I would also define a new type `StringSourceRange = Optional[List[Tuple[int, int]]]`

Reply:
`__future__` and friends are only available in Python >= 3.7 . No luck :P

Will see what I can do for the Ranges.

Reply:
I believe this: 

```
Range = Tuple[int, int]
ranges: Optional[List[Range]]
```

to be quite readable. As for the parse function, in theory I could have a `Dict[str, Any]`, I just decided to specify it a little bit. The parse function(s) will be the only place where we will have to deal with native raw types (such as Dict) so I am OK with it being the way it is (also not I am not checking linting int that specific line).

If you want more security, we could check if the given data format is the correct format, but I think that since we are reading from a darwin file, that we can do without the validation, since I am assuming the file is well generated.